cpp/src/umap/umap.cuh

```diff
@@ -34,7 +34,7 @@ static const int TPB_X = 256;
 
 inline bool dispatch_to_uint64_t(int n_rows, int n_neighbors, int n_components)
 {
-  uint64_t nnz1 = static_cast<uint64_t>(n_rows) * n_neighbors;
+  uint64_t nnz1 = 2 * static_cast<uint64_t>(n_rows) * n_neighbors;
```

Comment:
Let's not slow down merge since this is necessary to fix an urgent bug, but could we add a comment (either now or as an immediate follow-up) explaining why this is the correct limit? Same for nnz2 below.